crates/e2e/tests/e2e/cow_amm.rs

```diff
@@ -595,3 +596,685 @@ async fn cow_amm_driver_support(web3: Web3) {
+        app_data: AppDataHash(APP_DATA),
+        fee_amount: 0.into(),
+        kind: OrderKind::Sell,
+        partially_fillable: false,
+        sell_token_balance: Default::default(),
+        buy_token_balance: Default::default(),
+    };
+
+    // Create the signature for the CoW AMM order
+    let signature_data = ethcontract::web3::ethabi::encode(&[
+        Token::Tuple(vec![
+            Token::Address(cow_amm_order.sell_token),
+            Token::Address(cow_amm_order.buy_token),
+            Token::Address(cow_amm_order.receiver.unwrap_or_default()),
+            Token::Uint(cow_amm_order.sell_amount),
+            Token::Uint(cow_amm_order.buy_amount),
+            Token::Uint(cow_amm_order.valid_to.into()),
+            Token::FixedBytes(cow_amm_order.app_data.0.to_vec()),
+            Token::Uint(cow_amm_order.fee_amount),
+            Token::FixedBytes(
+                hex::decode("f3b277728b3fee749481eb3e0b3b48980dbbab78658fc419025cb16eee346775")
+                    .unwrap(),
+            ), // sell order
+            Token::Bool(cow_amm_order.partially_fillable),
+            Token::FixedBytes(
+                hex::decode("5a28e9363bb942b639270062aa6bb295f434bcdfc42c97267bf003f272060dc9")
+                    .unwrap(),
+            ), // sell_token_source == erc20
+            Token::FixedBytes(
+                hex::decode("5a28e9363bb942b639270062aa6bb295f434bcdfc42c97267bf003f272060dc9")
+                    .unwrap(),
+            ), // buy_token_destination == erc20
+        ]),
+        Token::Tuple(vec![
+            Token::Uint(0.into()), // min_traded_token
+            Token::Address(oracle.address()),
+            Token::Bytes(oracle_data),
+            Token::FixedBytes(APP_DATA.to_vec()),
+        ]),
+    ]);
+
+    // Prepend CoW AMM address to the signature so the settlement contract knows
+    // which contract this signature refers to.
+    let signature = cow_amm
+        .address()
+        .as_bytes()
+        .iter()
+        .cloned()
+        .chain(signature_data)
+        .collect::<Vec<_>>();
+
+    // Create the commitment call for the pre-interaction
+    let cow_amm_commitment = {
+        let order_hash = cow_amm_order.hash_struct();
+        let order_hash = hashed_eip712_message(&onchain.contracts().domain_separator, &order_hash);
+        let commitment = cow_amm
+            .commit(ethcontract::Bytes(order_hash))
+            .tx
+            .data
+            .unwrap();
+        Call {
+            target: cow_amm.address(),
+            value: 0.into(),
+            calldata: commitment.0.to_vec(),
+        }
+    };
+
+    // Fund trader "bob" with DAI and approve allowance
+    dai.mint(bob.address(), to_wei(250)).await;
+    tx!(
+        bob.account(),
+        dai.approve(onchain.contracts().allowance, U256::MAX)
+    );
+
+    // todo: figure out why this is even required.
+    dai.mint(onchain.contracts().gp_settlement.address(), to_wei(30))
+        .await;
```

Comment:
The PR shouldn't be merged until this is resolved.

Reply:
Can't reproduce the revert when the settlement contract is not funded enough using a manual rpc call with either anvil or forked tenderly node. So weird. Tried different options with/without state overrides, different block numbers. All the same: the simulation passes, but not in the test...

Reply:
[`17292a6` (#3038)](https://github.com/cowprotocol/services/pull/3038/commits/17292a6296958143a451675566026b93e5168141) fixes this 🤷‍♂️